lib/wsdl.js

```diff
@@ -1363,6 +1363,13 @@ WSDL.prototype.xmlToObject = function(xml, callback) {
       obj = null;
     }
 
+    // SUPER HACK
+    // topObject is sometimes a String instead of an Object which caused a runtime error, 
+    // what would make the response not be parsed to Object returning instead a String
+    if (typeof topObject === 'string') {
```

Comment:
Instead of checking the type for the `topObject` here, wouldn't it make more sense to prevent `obj` from being a `String` when pushing it onto the `stack` within the `p.onopentag()` function here at line [wsdl.js/L1343] (https://github.com/tommywalsh34/node-soap/blob/74c5f278bba888535bc0d4cf95d6fe7e8d9e8932/lib/wsdl.js#L1343) ?